pkg/generate/app/sourcelookup.go

```diff
@@ -15,6 +16,11 @@ import (
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
+var (
+	argumentGit         = regexp.MustCompile("^(http://|https://|git@|git://).*(?:#([a-zA-Z0-9]*))?$")
+	argumentGitProtocol = regexp.MustCompile("^(git@|git://)")
```

Comment:
ok, so i'm back to the position that we should strip the # before checking for git-validity.